client/lib/deploy.rs

```diff
@@ -111,6 +117,22 @@ impl<'a> OutputKind<'a> {
             OutputKind::Stdout => Ok(Box::new(io::stdout())),
         }
     }
+
+    pub(super) fn commit(self) -> Result<()> {
+        match self {
+            OutputKind::File { path, tmp_path, .. } => match fs::rename(tmp_path.clone(), path) {
+                Ok(_) => Ok(()),
+                Err(error) => Err(Error::IoError {
+                    context: format!(
+                        "Could not move tmp file {} to destination {}",
+                        tmp_path, path
+                    ),
+                    error,
+                }),
+            },
```

Comment:
Since `fs::rename()` returns a `Result<(), io::Error>`, we don't need to change the `Ok` type here - just mapping the `Err` type is enough.  Also, we don't need to clone `tmp_path`:

```suggestion
            OutputKind::File { path, tmp_path, .. } => {
                fs::rename(&tmp_path, path).map_err(|error| Error::IoError {
                    context: format!(
                        "Could not move tmp file {} to destination {}",
                        tmp_path.display(),
                        path
                    ),
                    error,
                })
            }
```